nilearn/decomposition/tests/test_canica.py

```diff
@@ -280,6 +280,19 @@ def test_with_globbing_patterns_with_single_subject(mask_img, tmp_path):
     assert components_img.shape, check_shape
 
 
+def test_with_globbing_patterns_with_single_subject_path(mask_img, tmp_path):
```

Comment:
we may need to add more checks like this that our public API can handle Path objects